### PR TITLE
update stellar-sdk and add keystore support for browser storage

### PR DESCRIPTION
**Pull request by piyalbasu**
This allows us to use Chrome or Firefox browser storage, which is necessary for Freighter:

https://developer.chrome.com/docs/extensions/reference/storage/
https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/storage

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -67,7 +67,7 @@
     "prettier": "^1.17.0",
     "regenerator-runtime": "^0.13.3",
     "sinon": "^7.3.1",
-    "stellar-sdk": "^9.0.1",
+    "stellar-sdk": "^10.1.0",
```
Comment (piyalbasu) [inline package.json]: I had some issues with npm linking because this was an outdated version

File: src/plugins/BrowserStorageFacade.ts [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+export interface BrowserStorageConfigParams {
+  prefix?: string;
+  storage: any;
+}
+
+/**
+ * Facade for `BrowserStorageKeyStore` encapsulating the access to the actual
+ * browser storage
+ */
```
Comment (piyalbasu) [inline src/plugins/BrowserStorageFacade.ts:11]: this file is very similar to `LocalStorageFacade`. This just takes into account browser storage's slightly different API

File: src/plugins/BrowserStorageFacade.ts [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+  public removeKey(id: string) {
+    this.check();
+    return this.storage !== null
+      ? this.storage.remove(`${this.prefix}:${id}`)
+      : null;
+  }
+
+  public async getAllKeys() {
+    this.check();
+    const regexp = RegExp(`^stellarkeys\\:(.*)`);
```
Comment (quietbits) [inline src/plugins/BrowserStorageFacade.ts:58]: We could make `stellarkeys` a const to avoid typos.

File: src/plugins/BrowserStorageKeyStore.ts [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+      this.keyStore.configure(params);
+      return Promise.resolve();
+    } catch (e) {
+      return Promise.reject(e);
+    }
+  }
+
+  public async storeKeys(keys: EncryptedKey[]) {
+    // We can't store keys if they're already there
+    const invalidKeys: EncryptedKey[] = [];
```
Comment (quietbits) [inline src/plugins/BrowserStorageKeyStore.ts:52]: `invalidKeys` sounds like there is something wrong with them. Maybe `usedKeys` or something like that would be a better name? :thinking:

**Review by quietbits — APPROVED** [review]

File: src/plugins/BrowserStorageFacade.ts [hunk 5]
```diff
@@ -0,0 +1,88 @@
+import { EncryptedKey } from "../types";
+
+export interface BrowserStorageConfigParams {
+  prefix?: string;
+  storage: any;
```
Comment (acharb) [inline src/plugins/BrowserStorageFacade.ts:5]:
```suggestion
  storage: Storage;
```

**Review by acharb — APPROVED** [review]
lgtm! had 1 comment

File: src/plugins/BrowserStorageFacade.ts (hunk 5 above)
Comment (piyalbasu) [reply src/plugins/BrowserStorageFacade.ts]: Ah, good catch - I forgot to add this interface. It's actually not `Storage`, which is the interface for localStorage. I added it using the spec here: https://developer.chrome.com/docs/extensions/reference/storage/

**Pull request merged**